source/client-side-encryption/client-side-encryption.rst

```diff
@@ -597,6 +631,66 @@ message.
 
 Return "access_token" as the credential.
 
+
+Obtaining an Access Token for Azure Key Vault
+`````````````````````````````````````````````
+
+.. versionadded:: 1.12.0 2022/08/16
+
+Virtual machines running on the Azure platform have one or more *Managed
+Identities* associated with them. From within the VM, an identity can be used by
+obtaining a access token via HTTP from the *Azure Instance Metadata Service*
+(IMDS). `See this documentation for more information`__
+
+__ https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http
+
+.. default-role:: math
+
+If an Azure VM has more than one managed identity, requesting an access token
+requires additional query parameters |--| `objectId`, `clientId`, and `miResId`
+|--| that disambiguate the request. We'll refer to these parameters as `M`. If
+the VM only has one managed identity, `M` may be omitted.
+
+The below steps should be taken:
+
+1. Let `U` be a new URL, initialized from the URL string
+   :ts:`"http://169.254.169.254/metadata/identity/oauth2/token"`
+2. Add a query parameter ``api-version=2018-02-01`` to `U`.
+3. Add a query parameter ``resource=http://vault.azure.com/`` to `U`.
+4. If managed identity parameters `M` are provided, add query parameters
+   ``object_id``, ``client_id``, and ``mi_res_id`` to `U` with the corresponding
+   values from `M`
+
+5. Prepare an HTTP GET request `Req` based on `U`.
+
+   .. note:: All query parameters on `U` should be appropriately percent-encoded
+
+6. Add an HTTP header ``Metadata: true`` to `Req`.
+
+7. Let `T` be a duration initialized to `0sec`.
+8. Issue `Req` to the Azure IMDS server ``168.254.169.254:80``. Let `Resp` be
+   the response from the server.
+9. If the `Resp_{status} >= 500`, wait for one second, then go to step ``8``.
+10. If `T < 30sec` AND `Resp_{status} = 404` OR `Resp_{status} = 429`, wait for
+    the duration specified by `T`, then set `T = (T * 2) + 2sec`, then go to
+    step ``8``.
+11. If `Resp_{status} ≠ 200`, obtaining the access token has failed, and the
+    HTTP response body of `Resp` encodes information about the error that
+    occurred. Return an error instead of an access token.
+12. Otherwise, let ``J`` be the JSON document encoded in the HTTP response body
+    of `Resp`.
+13. The result access token `T` is given as the ``access_token`` string property
+    of ``J``. Return `T` as the resulting access token.
```

Comment:
```suggestion
13. The result access token `Token` is given as the ``access_token`` string property
    of ``J``. Return `Token` as the resulting access token.
```

To avoid redeclaring `T`.